cdap-data-fabric/src/main/java/co/cask/cdap/spi/data/sql/jdbc/DataSourceInstantiator.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.spi.data.sql.jdbc;
+
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+import co.cask.cdap.common.conf.SConfiguration;
+import co.cask.cdap.common.lang.DirectoryClassLoader;
+import com.google.common.base.Throwables;
+import com.google.inject.Inject;
+import org.apache.commons.dbcp2.ConnectionFactory;
+import org.apache.commons.dbcp2.DriverManagerConnectionFactory;
+import org.apache.commons.dbcp2.PoolableConnection;
+import org.apache.commons.dbcp2.PoolableConnectionFactory;
+import org.apache.commons.dbcp2.PoolingDataSource;
+import org.apache.commons.pool2.ObjectPool;
+import org.apache.commons.pool2.impl.GenericObjectPool;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.sql.Driver;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.util.Map;
+import java.util.Properties;
+import java.util.function.Supplier;
+import javax.sql.DataSource;
+
+/**
+ * Class to instantiate the {@link DataSource} for the sql related structured table.
+ */
+public class DataSourceInstantiator implements Supplier<DataSource> {
+  private static final Logger LOG = LoggerFactory.getLogger(DataSourceInstantiator.class);
+
+  private final CConfiguration cConf;
+  private final SConfiguration sConf;
+
+  private DataSource dataSource;
+
+  @Inject
+  public DataSourceInstantiator(CConfiguration cConf, SConfiguration sConf) {
+    this.cConf = cConf;
+    this.sConf = sConf;
+  }
+
+  @Override
+  public DataSource get() {
+    if (dataSource != null) {
+      return dataSource;
+    }
+
+    if (cConf.get(Constants.Dataset.DATA_STORAGE_IMPLEMENTATION).equals(Constants.Dataset.DATA_STORAGE_NOSQL)) {
```

Comment:
To make this check generic, make it not equals to `Constants.Dataset.DATA_STORAGE_SQL`